src/main/java/com/example/onlinebookstore/dto/user/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,26 @@
+package com.example.onlinebookstore.dto.user;
+
+import com.example.onlinebookstore.validation.FieldMatch;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+@FieldMatch
+public class UserRegistrationRequestDto {
+    @NotNull
+    @Email
+    private String email;
+    @NotNull
+    @Length(min = 6, max = 20)

```

Comment:
Better use `Size` annotation, to be consistent. `Length` has the same meaning, but it's hibernate specific annotation.